server/server.go

```diff
@@ -249,6 +249,30 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 	}
 
 	if s.cfg.Socket != "" {
+
+		// Try to cleanup the socket file
+		sockStat, err := os.Stat(s.cfg.Socket)
```

Comment:
Maybe we could do the cleanup after `net.Listen("unix", s.cfg.Socket);` failed.

Reply:
So something like
```go
for i := 0; i < 2; i++ {
    if s.socket, err = net.Listen("unix", s.cfg.Socket); err != nil {
        if i > 0 {
            return nil, errors.Trace(err)
        } else {
            // Try to cleanup socket file
        }
    } else {
        break
    }
}
```

Doing this before the `net.Listen()` looks cleaner to me. What would be the benefit of doing this once it has failed?

Reply:
We don't have any overhead if it succeeds. Not a big benefit, but it is my preference

Reply:
@xhebox I prefer to cleanup the stale socket first. Otherwise the logging scenario is not clear, because if the `net.Listen` fails you probably want to log it, but only on `i > 0` scenarios.

Reply:
OK, anyway, it is fine.